src/__test__/App.test.js

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import store from '../redux/store';
+
+import App from '../App';
+import Homepage from '../components/pages/Homepage';
+import Details from '../components/pages/Details';
+
+jest.mock('../components/pages/Homepage');
+jest.mock('../components/pages/Details');
+
+describe('Testing my App.js with pages', () => {
+  test('should render the Header and Layout components', () => {
+    Homepage.mockImplementation(() => <h1>WELCOME TO AFRICA</h1>);
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('WELCOME TO AFRICA');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Homepage', () => {
+    Homepage.mockImplementation(() => <h1>Hello From Home Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Hello From Home Page');
+    expect(txt).toBeInTheDocument();
+  });
+
+  test('should render the Details Page', () => {
+    Details.mockImplementation(() => <h1>Hello From Details Page</h1>);
+    render(
+      <MemoryRouter initialEntries={['/details']}>
+        <Provider store={store}>
+          <App />
+        </Provider>
+      </MemoryRouter>,
+    );
+    const txt = screen.getByText('Hello From Details Page');
+    expect(txt).toBeInTheDocument();
+  });
+});
```

Comment:
- [ ] As requested by the previous reviewer, kindly make sure to solve the `App.test.js` because at this moment it is failing. Please fix it and make sure all tests are passing.

   
   ![image](https://user-images.githubusercontent.com/87197412/219016705-24201b65-e02e-48c7-952c-95ddb7e6aa5b.png)

   ![image](https://user-images.githubusercontent.com/87197412/219016803-3bd56862-b9d9-4bd4-9bf7-34640fd6afcf.png)

   ![image](https://user-images.githubusercontent.com/87197412/219017073-730e2d9f-9b81-4dde-87f5-2a96d79c4a6e.png)

   ![image](https://user-images.githubusercontent.com/87197412/219017147-e25be12b-4b88-48ff-9d23-3dba5fe365f1.png)